lib/resourcemerge/core.go

```diff
@@ -253,6 +253,18 @@ func EnsureServicePorts(modified *bool, existing *[]corev1.ServicePort, required
 	}
 }
 
+func EnsureServiceType(modified *bool, existing *corev1.ServiceType, required corev1.ServiceType) {
+	// if we have no required, then we don't care what someone else has set
+	if required == "" {
```

Comment:
can we require [the default value of `ClusterIP`][1]?  Because I don't think we want to let cluster admins come in and say "actually, I think this should be `NodePort`" or anything.

[1]: https://github.com/kubernetes/api/blob/9b64426eca51a74faa7cc9bd732a533d339c69c2/core/v1/types.go#L4149